src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.cs

```diff
@@ -1599,12 +1609,18 @@ protected override void OnPointerReleased(PointerRoutedEventArgs args)
 			}
 		}
 
-#if !__ANDROID__ && !__IOS__ // ScrollContentPresenter.[Horizontal|Vertical]Offset not implemented on Android and iOS
+		// On WASM, we could choose to scroll in the managed layer and suppress the native scrolling
+		// but it can lead to some chaotic scenarios where it's really difficult to reconcile the
+		// numbers between ScrollViewer and ScrollContentPresenter, so we choose to keep the scrolling native
+#if UNO_HAS_MANAGED_SCROLL_PRESENTER || UNO_REFERENCE_API
```

Comment:
It doesn't feel right to have the override for reference but not for Wasm. Hopefully, the CI should detect this and fail.